CONTRIBUTING.md

```diff
@@ -190,6 +190,45 @@ The system follows a linear event pipeline:
 
 - **Comments**: Only explain _why_ something is done, not _what_ is done.
 
+### **GitHub Pages Documentation Standards**
+
+When working on the GitHub Pages documentation site (`docs/`):
+
+- **File Organization**:
+  - `index.html` - Minimal HTML structure only
+  - `style.css` - All styles with CSS variables and full reset
+  - `app.js` - All JavaScript logic and configuration
+  
+- **CSS Guidelines**:
+  - Use full CSS reset: `* { all: initial; }`
+  - Define all values as CSS variables in `:root`
+  - Prefer `rem` units over `px` for scalability
+  - Group related variables (colors, spacing, typography, layout)
+  - Use HSL color values with semantic naming (e.g., `--white`, `--gray-500`, then semantic like `--color-bg-primary`)
+  - Implement dark mode support with `@media (prefers-color-scheme: dark)`
+  - Prefer `system-ui, "Helvetica Neue"` for font family
+  
+- **JavaScript Guidelines**:
+  - Keep configuration data-driven (see `documentationSections` array)
+  - Sidebar navigation should be dynamically generated from configuration
+  - Maintain single source of truth for section definitions
+  - Use module imports for external dependencies
+  - Use import maps in HTML for cleaner dependency management
+  - Use `main()` as the primary entry point function
+  - Use self-invoking anonymous functions for initialization
+  
+- **HTML Guidelines**:
+  - Keep HTML minimal - structure only, no inline styles
+  - Link external CSS and JS files
+  - Empty sidebar element to be populated by JavaScript
+  - Use semantic HTML5 elements
+  - Include import maps for module dependencies
+  
+- **Future Enhancements** (marked for consideration):
```

Comment:
can you make an issue for these instead? and include turning type checking on in app.js with jsdocs